pulsar-broker/src/main/java/org/apache/pulsar/broker/service/BrokerService.java

```diff
@@ -1182,7 +1182,9 @@ private void updateTopicMessageDispatchRate() {
                 if (topic instanceof PersistentTopic) {
```

Comment:
Instead of the `instanceof` check we could just add the `updateDispatchRate()` in the `Topic` interface with empty implementation for non-persistent topics.